src/odemis/acq/test/millmng_test.py

```diff
@@ -0,0 +1,332 @@
+                                            roi=(0.5, 0.5, 0.8, 0.8),
+                                            pixel_size=(3.41796875e-08, 3.41796875e-08), beam_angle=self.beam_angle,
+                                            duration=20,
+                                            dc_roi=UNDEFINED_ROI, dc_period=10, dc_dwell_time=10e-6, dc_current=self.probe_current)
+
+        milling_setting_2 = MillingSettings(name='rough_milling_2', current=self.probe_current, horizontal_fov=35e-6,
+                                            roi=(0.5, 0.5, 0.8, 0.8),
+                                            pixel_size=(3.41796875e-08, 3.41796875e-08), beam_angle=self.beam_angle,
+                                            duration=20,
+                                            dc_roi=UNDEFINED_ROI, dc_period=10, dc_dwell_time=10e-6,
+                                            dc_current=self.probe_current)
+
+        millings = [milling_setting_1, milling_setting_2]
+
+        task = MillingRectangleTask(model.InstantaneousFuture(), millings, self.sites, self.feature_post_status,
+                                    self.acq_streams, self.ion_beam,
+                                    self.sed, self.stage, self.aligner)
+
+        # Time estimate in the beginning
+        time_estimated = task.estimate_milling_time(site_idx=0)
+        min_milling_time = milling_setting_1.duration.value + milling_setting_2.duration.value
+        self.assertGreaterEqual(time_estimated, min_milling_time,
+                                "estimated time is less than the actual time set to mill two features")
+
+        # Time estimate after 1 complete milling
+        time_estimated = task.estimate_milling_time(site_idx=1)
+        min_milling_time = milling_setting_2.duration.value
+        self.assertGreaterEqual(time_estimated, min_milling_time,
+                                "estimated time is less than the actual time while milling the second feature")
+
+        total_est_time_without_drift = task.estimate_milling_time(site_idx=0)
+
+        # B. With Drift
+        milling_setting_1 = MillingSettings(name='rough_milling_1', current=self.probe_current, horizontal_fov=35e-6,
+                                            roi=(0.5, 0.5, 0.8, 0.8),
+                                            pixel_size=(3.41796875e-08, 3.41796875e-08), beam_angle=self.beam_angle,
+                                            duration=20,
+                                            dc_roi=(0, 0.3, 0.4, 0.7), dc_period=10, dc_dwell_time=10e-6,
+                                            dc_current=self.probe_current)
+
+        milling_setting_2 = MillingSettings(name='rough_milling_2', current=self.probe_current, horizontal_fov=35e-6,
+                                            roi=(0.5, 0.5, 0.8, 0.8),
+                                            pixel_size=(3.41796875e-08, 3.41796875e-08), beam_angle=self.beam_angle,
```

Comment:
Please try some milling settings with non-square pixel_size. Typically, for the MIMAS the Y should be smaller that X (like, 6 times smaller). However, it'd be good to try also when X is smaller than Y.